pkg/kubelet/kubelet.go

```diff
@@ -1668,11 +1753,18 @@ func (kl *Kubelet) tryUpdateNodeStatus() error {
 	// cAdvisor locally, e.g. for test-cmd.sh, and in integration test.
 	info, err := kl.GetCachedMachineInfo()
 	if err != nil {
-		glog.Errorf("error getting machine info: %v", err)
+		// TODO(roberthbailey): This is required for test-cmd.sh to pass.
```

Comment:
That sounds reasonable and it'll allow us to keep data we knew to be true.

It is important to note that under normal circumstances cAdvisor is
running, I have yet to see a situation in production where it is not. These
edge cases happen in testing where we mock or stub out different parts of
the Kubelet.

On Fri, May 15, 2015 at 10:08 AM, Clayton Coleman <notifications@github.com>
wrote:

> In pkg/kubelet/kubelet.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6949#discussion_r30426565>
> :
>
> > @@ -1668,11 +1753,18 @@ func (kl *Kubelet) tryUpdateNodeStatus() error {
> >  	// cAdvisor locally, e.g. for test-cmd.sh, and in integration test.
> >  	info, err := kl.GetCachedMachineInfo()
> >  	if err != nil {
> > -		glog.Errorf("error getting machine info: %v", err)
> > +		// TODO(roberthbailey): This is required for test-cmd.sh to pass.
>
> Anyone can call the node api. If cadvisor can't report anything, and the
> node already has something, should the node overwrite the capacity? I would
> argue it should set zero if nothing is set, but if it's explicitly set
> (perhaps from the last time) it shouldn't be reset. Not a massively strong
> feeling, but if cadvisor is dead I don't know that we should be making the
> node completely unschedulable (by forcing capacity to 0)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/6949/files#r30426565>
> .
>